handlers/search.go

```diff
@@ -134,145 +133,124 @@ func SearchAhead(w http.ResponseWriter, r *http.Request) {
-			if err != nil {
-				logger.Errorf("error reading result data: %v", err)
-				http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-				return
-			}
 		} else if thresholdHexLikeRE.MatchString(search) {
 			err = db.ReaderDb.Select(result, `SELECT validatorindex AS index, pubkeyhex as pubkey FROM validators WHERE pubkeyhex LIKE LOWER($1 || '%')`, search)
-			if err != nil {
-				logger.Errorf("error reading result data: %v", err)
-				http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-				return
-			}
 		} else {
 			err = db.ReaderDb.Select(result, `
 			SELECT validatorindex AS index, pubkeyhex AS pubkey
 			FROM validators
 			LEFT JOIN validator_names ON validators.pubkey = validator_names.publickey
 			WHERE LOWER(validator_names.name) LIKE LOWER($1)
 			ORDER BY index LIMIT 10`, search+"%")
-			if err != nil {
-				logger.Errorf("error reading result data: %v", err)
-				http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-				return
-			}
+		}
+		if err != nil {
+			logger.Errorf("error reading result data: %v", err)
+			http.Error(w, "Internal server error", http.StatusInternalServerError)
+			return
 		}
 	case "validators_by_pubkey":
 		result = &types.SearchAheadPubkeyResult{}
-		if thresholdHexLikeRE.MatchString(search) {
-			// Find the validators that have made a deposit but have no index yet and therefore are not in the validators table
-			err = db.ReaderDb.Select(result, `
-				SELECT DISTINCT
-					ENCODE(eth1_deposits.publickey, 'hex') AS pubkey
-					FROM eth1_deposits
-					LEFT JOIN validators ON validators.pubkey = eth1_deposits.publickey
-					WHERE validators.pubkey IS NULL AND ENCODE(eth1_deposits.publickey, 'hex') LIKE ($1 || '%')`, search)
-			if err != nil {
-				logger.Errorf("error reading result data: %v", err)
-				http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-				return
-			}
+		if !thresholdHexLikeRE.MatchString(search) {
+			break
 		}
-	case "indexed_validators_by_eth1_addresses":
-		result, err = FindValidatorIndicesByEth1Address(search)
+		// Find the validators that have made a deposit but have no index yet and therefore are not in the validators table
+		err = db.ReaderDb.Select(result, `
+		SELECT DISTINCT
+			ENCODE(eth1_deposits.publickey, 'hex') AS pubkey
+			FROM eth1_deposits
+			LEFT JOIN validators ON validators.pubkey = eth1_deposits.publickey
+			WHERE validators.pubkey IS NULL AND ENCODE(eth1_deposits.publickey, 'hex') LIKE ($1 || '%')`, search)
 		if err != nil {
-			http.Error(w, "Internal server error", http.StatusServiceUnavailable)
+			logger.Errorf("error reading result data: %v", err)
+			http.Error(w, "Internal server error", http.StatusInternalServerError)
+			return
+		}
+	case "indexed_validators_by_eth1_addresses":
+		if utils.IsValidEnsDomain(search) || utils.IsEth1Address(search) {
+			result, err = FindValidatorIndicesByEth1Address(search)
```

Comment:
The error handling is not consistent. There is no error handling in this branch so https://github.com/gobitfly/eth2-beaconchain-explorer/blob/71334ade1670beeec7eae32b60c7b73b7e7ae9dd/handlers/search.go#L363-L367 would be used.
This seems valid but why is there error handling for example here too: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/71334ade1670beeec7eae32b60c7b73b7e7ae9dd/handlers/search.go#L293-L297

Please share your thoughts.

Reply:
My thoughs are as follows: We should probably be consistent, so I refactored the search so that every error is handled by the handler after the switch. Ptal